sonar-db/src/test/java/org/sonar/db/permission/template/PermissionTemplateCharacteristicDaoTest.java

```diff
@@ -179,4 +180,30 @@ public void fail_update_if_id_is_null() {
       .setCreatedAt(123_456_789L)
       .setUpdatedAt(2_000_000_000L));
   }
+
+  @Test
+  public void delete_by_permission_template_id() throws Exception {
+    underTest.insert(dbSession, new PermissionTemplateCharacteristicDto()
+      .setPermission(UserRole.USER)
+      .setTemplateId(1L)
+      .setWithProjectCreator(true)
+      .setCreatedAt(123_456_789L)
+      .setUpdatedAt(2_000_000_000L));
+    underTest.insert(dbSession, new PermissionTemplateCharacteristicDto()
+      .setPermission(UserRole.USER)
+      .setTemplateId(2L)
+      .setWithProjectCreator(true)
+      .setCreatedAt(123_456_789L)
+      .setUpdatedAt(2_000_000_000L));
+    dbSession.commit();
+
+    assertThat(underTest.selectByTemplateIds(dbSession, asList(1L))).hasSize(1);
+    assertThat(underTest.selectByTemplateIds(dbSession, asList(1L, 2L))).hasSize(2);
+
+    underTest.deleteByTemplateId(dbSession, 1L);
+    dbSession.commit();
```

Comment:
You right, I don't know why it was failing without the commit when I've begun to write this test.